### PR TITLE
Fix `comments-time-machine-links` on first conversation comment

### PR DESCRIPTION
**Pull request by cheap-glitch**
`comments-time-machine-links` broke on issue/PR bodies (the first comment of the conversation):

```text
❌ comments-time-machine-links 22.2.13 → TypeError: menu.closest(...) is null
	init moz-extension://6ffeab5e-837d-450e-9fa5-fe3ec6a7ee9f/build/refined-github.js:6345
	runFeature moz-extension://6ffeab5e-837d-450e-9fa5-fe3ec6a7ee9f/build/refined-github.js:3045
```

This PR also fixes the feature for minimized comments inside review threads.

## Test URLs

https://togithub.com/refined-github/refined-github/pull/5422

It should work on:
- first comment (PR body)
- regular comments
- "main" review comments
- review thread comments
- review comments in minimized thread
- minimized review thread comments

## Screenshot

Conversation body (i.e. first comment):
image

Minimized comment in review thread:
image

### CODE REVIEW

File: source/features/comments-time-machine-links.tsx [hunk 1]
```diff
@@ -110,7 +110,10 @@ function init(): void {
 	const commentPopupMenus = select.all('.js-reaction-popover-container ~ details:last-child:not(.rgh-time-machine-links)');
 	for (const menu of commentPopupMenus) {
 		menu.classList.add('rgh-time-machine-links');
-		const timestamp = menu.closest('.js-comment:not(.timeline-comment-group), .js-timeline-item')!.querySelector('relative-time')!.attributes.datetime.value;
+		// The timestamp of main review comments isn't in their header but in the timeline event above #5423
+		const timestamp = menu.closest('.js-comment:not([id^="pullrequestreview-"]), .js-timeline-item')!
```
Comment (cheap-glitch) [inline source/features/comments-time-machine-links.tsx:114]: `:not(.timeline-comment-group)` was here to exclude main review comments, but now it also excludes the first comment of the issue/PR.

Fixed by making the selector more specific + added an explanatory comment for future reference.

File: source/github-events/on-new-comments.ts [hunk 2]
```diff
@@ -7,6 +7,8 @@ const delegates = new Set<delegate.Subscription>();
 const observer = new MutationObserver(run);
 
 function run(): void {
+	console.log(new Date().toLocaleString().split(',').pop()!.trim());
```
Comment (yakov116) [inline source/github-events/on-new-comments.ts:10]: Logged successful 😄

File: source/features/comments-time-machine-links.tsx [hunk 3]
```diff
@@ -107,10 +107,13 @@ function addDropdownLink(menu: HTMLElement, timestamp: string): void {
 }
 
 function init(): void {
-	const commentPopupMenus = select.all('.js-reaction-popover-container ~ details:last-child:not(.rgh-time-machine-links)');
+	const commentPopupMenus = select.all('.timeline-comment-actions > details:last-child:not(.rgh-time-machine-links)');
```
Comment (cheap-glitch) [inline source/features/comments-time-machine-links.tsx:110]: Minimized comments inside review threads don't have a reaction dropdown :man_shrugging: 

image

File: source/github-events/on-new-comments.ts (hunk 2 above)
Comment (cheap-glitch) [reply source/github-events/on-new-comments.ts:10]: (｢•-•)｢ ︵ ┻━┻

That's what I get for using `git commit --all` every time

File: source/features/comments-time-machine-links.tsx [hunk 4]
```diff
@@ -107,10 +107,13 @@ function addDropdownLink(menu: HTMLElement, timestamp: string): void {
...
 	for (const menu of commentPopupMenus) {
 		menu.classList.add('rgh-time-machine-links');
-		const timestamp = menu.closest('.js-comment:not(.timeline-comment-group), .js-timeline-item')!.querySelector('relative-time')!.attributes.datetime.value;
+		// The timestamp of main review comments isn't in their header but in the timeline event above #5423
+		const timestamp = menu.closest('.js-comment:not([id^="pullrequestreview-"]), .js-timeline-item')!
```
Comment (fregante) [inline source/features/comments-time-machine-links.tsx:114]:
```suggestion
		const timestamp = menu
			.closest('.js-comment:not([id^="pullrequestreview-"]), .js-timeline-item')!
```

**Review by fregante — APPROVED** [review]

**Comment by kidonng** [comment]
Just some thoughts:

Ideally we should have general handlers for comments so instead of every feature maintaining their own selectors and status checking, we have just one and features register actions to specific types of comments.

Same for files/commits or other stuff that has many entries.

**Comment by kidonng** [comment]
> I thought about that before, but it may be tricky because there's a lot of subtle differences between similar-looking selectors and the way they're used in each feature.

🤷 Maybe I will submit a POC when I got time

**Pull request merged**